salt/states/win_task.py

```diff
@@ -0,0 +1,483 @@
+# You can add or remove tasks.
+
+# Import Python libs
+import copy
+import logging
+import time
+
+# Import salt libs
+import salt.utils.data
+import salt.utils.platform
+import salt.utils.dateutils
+
+# Import 3rd-party libs
+try:
+    import pywintypes
+except ImportError:
+    pass
+
+
+ACTION_PARTS = {'Execute': ['cmd'],
+                'Email': ['from', 'to', 'cc', 'server'],
+                'Message': ['title', 'message']}
+
+OPTIONAL_ACTION_PARTS = {'start_in': '',
+                         'arguments': ''}
+
+TRIGGER_PARTS = {'Event': ['subscription'],
+                 'Once': [],
+                 'Daily': ['days_interval'],
+                 'Weekly': ['days_of_week', 'weeks_interval'],
+                 'Monthly': ['months_of_year', 'days_of_month', 'last_day_of_month'],
+                 'MonthlyDay': ['months_of_year', 'weeks_of_month', 'days_of_week'],
+                 'OnIdle': [],
+                 'OnTaskCreation': [],
+                 'OnBoot': [],
+                 'OnLogon': [],
+                 'OnSessionChange': ['state_change']}
+
+OPTIONAL_TRIGGER_PARTS = {'trigger_enabled': True,
+                          'start_date': time.strftime('%Y-%m-%d'),
+                          'start_time': time.strftime('%H:%M:%S'),
+                          'end_date': None,
+                          'end_time': "00:00:00",
+                          'random_delay': False,
+                          'repeat_interval': None,
+                          'repeat_duration': None,
+                          'repeat_stop_at_duration_end': False,
+                          'execution_time_limit': '3 days',
+                          'delay': False}
+
+OPTIONAL_CONDITIONS_PARTS = {'ac_only': True,
+                             'run_if_idle': False,
+                             'run_if_network': False,
+                             'start_when_available': False}
+
+OPTIONAL_SETTINGS_PARTS = {'allow_demand_start': True,
+                           'delete_after': False,
+                           'execution_time_limit': '3 days',
+                           'force_stop': True,
+                           'multiple_instances': 'No New Instance',
+                           'restart_interval': False,
+                           'stop_if_on_batteries': True,
+                           'wake_to_run': False}
+
+TASK_PARTS = {'actions': {'parts': ACTION_PARTS, 'optional': OPTIONAL_ACTION_PARTS},
+              'triggers': {'parts': TRIGGER_PARTS, 'optional': OPTIONAL_TRIGGER_PARTS},
+              'conditions': {'parts': {}, 'optional': OPTIONAL_CONDITIONS_PARTS},
+              'settings': {'parts': {}, 'optional': OPTIONAL_SETTINGS_PARTS}}
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'task'
+
+
+def __virtual__():
+    '''
+    Load only on minions running on Windows and with task Module loaded.
+    '''
+    if not salt.utils.platform.is_windows() or 'task.list_tasks' not in __salt__ or\
+        'pywintypes' not in globals():
```

Comment:
I'm not really keen on this globals lookup.

```python
try:
    import pywintypes
except ImportError:
    pywintypes = None
```
or
```python

try:
    import pywintypes
    HAS_PYWINTYPES = True
except ImportError:
    HAS_PYWINTYPES = False
```
And then use one of the above approaches instead of `'pywintypes' not in globals()`